contracts/bprotocol/Import.sol

```diff
@@ -0,0 +1,227 @@
+                           address[] calldata collateralUnderlying,
+                           address[] calldata cTokenDebt,
+                           address[] calldata debtUnderlying,
+                           uint      ethFlashLoan) external {
+
+        require(cTokenCollateral.length == collateralUnderlying.length, "collateral-length-missmatch");
+        require(cTokenDebt.length == debtUnderlying.length, "debt-length-missmatch");
+        require(mul_(ethFlashLoan, 100) <= mul_(address(this).balance, 101), "flashloan-fees-over-1%");
+
+        address account = tx.origin;
+        address avatar = registry.getAvatar(account);
+
+        // mint ETH on B
+        uint ethBalance = address(this).balance;
+        BEther(bETH).mintOnAvatar.value(ethBalance)(avatar);
+
+        uint[] memory originalDebt = new uint[](cTokenDebt.length);
+        for(uint i = 0 ; i < cTokenDebt.length ; i++) {
+          originalDebt[i] = ICToken(cTokenDebt[i]).borrowBalanceCurrent(account);
+        }
+
+        // borrow the original debt
+        borrowDebtOnB(cTokenDebt,
+                      originalDebt,
+                      avatar);
+
+        // repay all debt
+        repayAllDebt(cTokenDebt, debtUnderlying, originalDebt, account);
+
+        // redeem all non ETH collateral, deposit it in B
+        redeemAndDepositCollateral(cTokenCollateral,
+                                   collateralUnderlying,
+                                   account,
+                                   avatar);
+
+        BEther(bETH).redeemUnderlyingOnAvatar(avatar, ethBalance);
+        if(address(this).balance < ethFlashLoan) {
+          BEther(bETH).borrowOnAvatar(avatar, ethFlashLoan - address(this).balance);
+        }
+
+        msg.sender.transfer(address(this).balance);
+    }
+
+    // accept ETH transfers
+    function() external payable {}
+}
+
+contract FlashLoanLike {
+    function borrow(address _token, uint _amount, bytes calldata _data) external;
+}
+
+// user will call this contract
+// this contract will be upgraded over time, depending on available flash loan
+// sources
+contract FlashLoanImport {
+    using SafeERC20 for IERC20;
+    address constant ETH = 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+
+    function flashImport(address[] calldata cTokenCollateral,
+                         address[] calldata collateralUnderlying,
+                         address[] calldata cTokenDebt,
+                         address[] calldata debtUnderlying,
+                         address payable importer,
+                         uint            ethAmountToFlashBorrow,
+                         address payable flash) external {
+        require(cTokenDebt.length == debtUnderlying.length, "debt-length-missmatch");
+
+        if(address(this).balance < ethAmountToFlashBorrow) {
+          // this will invoke a recursive call
+          return FlashLoanLike(flash).borrow(ETH, ethAmountToFlashBorrow, msg.data);
```

Comment:
Is flash loan contract at lending pool support ETH address? or is it something different?

Reply:
in keeperdao they do.
we will have to change it if we use dydx. for now let's follow that.